src/main/java/com/google/api/generator/gapic/model/Message.java

```diff
@@ -80,6 +89,32 @@ public boolean hasResource() {
     return resource() != null;
   }
 
+  public void validateField(String fieldName, Map<String, Message> messageTypes) {
+    String[] subFields = fieldName.split("\\.");
```

Comment:
IMO, `String.split` is fine in simple and obvious cases, but make sure you understand [the pitfall](
https://errorprone.info/bugpattern/StringSplitter).

Reply:
Good to know, I can switch to Guava split.